packages/cli/src/worker-utils.ts

```diff
@@ -0,0 +1,165 @@
+import { dirname, relative } from 'path';
+import type { ResolvedProjectReference } from 'typescript';
+import typescript from 'typescript';
+import { Worker } from 'worker_threads';
+
+import { WorkerError } from './errors.js';
+import { success } from './logging.js';
+import { createProjectReferencesCompilerHost } from './project-references.js';
+import type { BuilderOptions } from './shared.js';
+import {
+  cleanOutputDirectory,
+  getBuildFunction,
+  getInitialProgram,
+} from './shared.js';
+import { getDefinedArray } from './utils.js';
+
+const { sys } = typescript;
+
+export type WorkerOptions = Omit<
+  BuilderOptions,
+  | 'compilerOptions'
+  | 'files'
+  | 'host'
+  | 'program'
+  | 'projectReferences'
+  | 'system'
+> & {
+  project: string;
+};
+
+export type GetBuildWorkerFunctionOptions = Omit<
+  WorkerOptions,
+  'baseDirectory' | 'name' | 'project' | 'tsConfig'
+> & {
+  parentBaseDirectory: string;
+};
+
+/**
+ * Get the build worker function.
+ *
+ * @param options - The options to use for the worker.
+ * @returns The build worker function.
+ */
+export function getBuildWorkerFunction(options: GetBuildWorkerFunctionOptions) {
+  /**
+   * Build a project reference using a worker thread.
+   *
+   * @param projectReference - The project reference to build.
+   */
+  return async function buildWorker(
```

Comment:
This or `workerBuild()` would be better:
```suggestion
  return async function buildInWorker(
```